main.go

```diff
@@ -340,6 +343,40 @@ func decodeFortDetails(ctx context.Context, sDec []byte) string {
 	return "Unknown fort type"
 }
 
+func decodeGetMapForts(ctx context.Context, sDec []byte) string {
+	decodedMapForts := &pogo.GetMapFortsOutProto{}
+	if err := proto.Unmarshal(sDec, decodedMapForts); err != nil {
+		log.Fatalln("Failed to parse", err)
+		return fmt.Sprintf("Failed to parse %s", err)
+	}
+
+	if decodedMapForts.Status != pogo.GetMapFortsOutProto_SUCCESS {
+		res := fmt.Sprintf(`GetMapFortsOutProto: Ignored non-success value %d:%s`, decodedMapForts.Status,
+			pogo.GetMapFortsOutProto_Status_name[int32(decodedMapForts.Status)])
+		return res
+	}
+
+	var outputString string
+	processedForts := 0
+
+	for _, fort := range decodedMapForts.Fort {
+		// when we miss, we check the gym, if again, we save it in cache for 5 minutes (in gym part)
+		status, output := decoder.UpdatePokestopRecordWithGetMapFortsOutProto(ctx, dbDetails, fort)
+		if !status {
+			status, output = decoder.UpdateGymRecordWithGetMapFortsOutProto(dbDetails, fort)
+		}
```

Comment:
I think it should add to the cache here not as magic inside the UpgradeGymRecord.

Reply:
getMapFortsCache would be available to whole of the module surely?
It's magic because the two methods do something different. One just tries an update on a pokestop, the other tries the update on the gym, but if it doesn't work it invisibly adds it to the cache